### DISCUSSION TITLE
Questions

### DISCUSSION THREAD

**Discussion by clumba** [TF2Autobot/tf2autobot#1593] [Q&A]
1) I want to accept weapons as payment, but I don't want to create listings on the site. Is it possible? 

2) If you specify intent=sell in the weapons listing settings, then when you receive a trade offer where two weapons are offered instead of scrap, the current logic rejects the offer (autoDecline set to false). Is there any way to make a switch to manual review instead of declining in order to approve/reject such offers manually?

**Comment by idinium96** [comment]
Hello,
There is already an option to set weapons as a currency.
https://github.com/TF2Autobot/tf2autobot/blob/89eb808ec2390f87a9349b02eb0cf9cf7dddd4a1/.example/options.json#L30-L33
And it's already set to `true` (enabled) by default.

<https://github.com/TF2Autobot/tf2autobot/wiki/Configure-your-options.json-file#--weapons-as-currency-->
![Screenshot_75](https://github.com/TF2Autobot/tf2autobot/assets/47635037/e2015cd3-bfa1-456a-a62a-711a236f60ce)

What weapons are deemed as currency?
All weapons that exist in TF2, except tradeable stock weapons, festive weapons, festivized weapons, weapons with attached spells, and the listed below:
https://github.com/TF2Autobot/node-tf2-schema/blob/fa4efa8ae42b1c7cc9707dc1802f4f05b59aa601/lib/schema.js#L1732-L1753

**Comment by clumba** [comment]
Yes, I read it. I will try to describe the problem more precisely. In my pricelist.json added an items (weapons) in this form: 
```
"513;6":{"article": "513;6","title":"Original","included":true,"auto price":false,"min":0,"max":99,"intent":1,"buy":{"keys":0,"metal":0.05},
"sell":{"keys":0,"metal":0.11}, 
```
And if someone tries to buy it with weapons, the bot declines the trade with the following message: 

> {"level":"detailed","message":"Offer #xxx from xxx status changed: Active -> Rejected (reason: GIVING_ITEMS_WITH_INTENT_SELL)","timestamp":"2023-07- 06T04:51:27.569Z"}
> {"Completion time":1826,"Level": "Debugging","Message":"It took 1826 ms to complete the offer","Offer ID":"xxx","Status":7,"Timestamp": "2023-07-06T04:51:27.570Z"}
> {"level":"information","message":"Message sent to xxx(xxx): /me Transaction #xxx with xxx was rejected. ❌ \n\nSummary (offer)\n• Requested: 0.11 ref (atomizer)\n• Proposed: 0.11 ref (Ullapool Caber, Bushwacka)\n──────────────────────\ n\end lists:\n 📜 _ITEMS_PRICES\n- Atomizer - 0.05 ref / 0.11 ref (manual),\n- Ullapool Caber - 0.05 ref/ 0.11 ref (manual),\n- Bushwacka - 0.05 ref / 0.11 ref (manual)\n\n 🔑 Key rate: 80.77 ref/80.88 ref (prices.tf)\n 💰 Net stock: 59.88 references (52 references, 18 recommendations, 17 waste) \n 🎒 Total products: 222/500\n ⏱ Time spent: \n- To process the offer: 1 second\n\n Version 5.8.1","timestamp": "2023-07-06T04:51:27.973Z"}
> {"level":"information","message":"The message was sent to an unknown (xxx - not a friend): /previously ❌ Oooh, nooo! The offer is no longer valid. Reason: The offer was rejected because you are trying to provide goods that I only want to sell.","timestamp":"2023-07-06T04:51:28.010Z"} 

Am I doing something wrong?

**Reply by idinium96** [reply]
I have never seen the `article`, `title`, and `included` keys on that pricelist.json. Are you sure you're using tf2autobot?

Also, why did you set it as intent=sell? Also, the selling price is 0.11, where as a currency, it should just be 0.05.

**Reply by clumba** [reply]
`I have never seen the article, title, and included keys on that pricelist.json. Are you sure you're using tf2autobot?`

Of course, I use tf2autobot. It was installed according to the instructions in the wiki. The syntax in the file is obtained when adding items via the command:
`!add item=Back Scratcher&intent=sell&sell.metal=0.11&buy.metal=0.05`

> Also, why did you set it as intent=sell?

Because I want to sell weapons. Should the parameter value be different?

> Also, the selling price is 0.11, where as a currency, it should just be 0.05.

As far as I understand, the value of the currency is indicated separately for cases of purchase and sale - as I indicated in the example above. Is there some kind of mistake?
